source/scpd/Cpp.d

```diff
@@ -235,6 +235,23 @@ extern(C++, (StdNS!())) struct vector (T, Alloc = allocator!T)
             return this._start[start .. end];
         }
 
+        public bool opEquals (in vector rhs) const pure nothrow @nogc @trusted
```

Comment:
BY value ? 👎

Reply:
It's three pointers though. Isn't it more efficient to pass three pointers instead of a pointer to three pointers and do double indirection that way? I'd have to look up an assembly book to confirm. :P

Reply:
Maybe (probably?) but I'm more concerned about passing C++ structs by value. Even though we don't currently implement the copy constructor that does the RC, we might in the future.

Reply:
Aha. That's a good point actually.

Reply:
I always thought you were using `ref` due to performance, I forgot about the mess D has with copy ctors/postblits/etc.